### PR TITLE
Refactors some borg inventory code and removes a bunch of unused references to borgs taking oxy/tox damage

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

This PR refactors borg inventory code - equipping and un-equipping items to slots, the process of how slots are broken and repaired, how modules are selected, etc. In the process, this fixes the use-items-in-broken-slots tech. Sorry, borg mains.

This PR also removes a few old, now deprecated references to cyborgs taking oxygen and toxin damage. There was a few locations that called adjustToxLoss and adjustOxyLoss of cyborgs, despite the fact that the procs were overridden and did nothing. The big performance change lies in the cyborg life proc - I removed the adjustOxyLoss in the life proc, because it was calling updateHealth every life tick, wasting processing.

Let me know if I overstepped myself in any areas, and accidentally cleaned something I shouldn't have.

**If there are any situations I missed in which a cyborg can take oxygen/tox damage, please let me know, so I can remove them too or adjust the PR accordingly.**

Probably fixes part of this issue, but not all of it #51299

## Why It's Good For The Game

Makes some of the code for cyborgs nicer, and improves the speed of the code.

before the refactor (~20 minutes in game with 10 cyborgs spawned):
image
after the refactor (~22 minutes in game with 10 cyborgs spawned):
image

## Changelog
:cl: Melbert
fix: You can no longer use broken cyborg slots by being fast at clicking
code: Removed a bunch of deprecated cyborg code involving taking toxin and oxygen damage
refactor: Refactored part of cyborg inventory code
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
Good PR

**Comment by ATH1909** [comment]
> In the process, this fixes the use-items-in-broken-slots tech. Sorry, borg mains.
Damn.

Still, I support this PR; you dun did good here, MrMelbert.

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 1]
```diff
@@ -1,82 +1,233 @@
...
+		if(3)
+			if(!(disabled_modules & BORG_MODULE_THREE_DISABLED))
+				return FALSE
+		
+			inv3.icon_state = "inv3"
+			disabled_modules &= ~BORG_MODULE_THREE_DISABLED
+
+	to_chat(src, "<span class='notice'>ERROR CLEARED: Module [module_num] back online.</span>")
+
+	return TRUE
```
Comment (bobbahbrown) [inline code/modules/mob/living/silicon/robot/inventory.dm:216]: would it be feasible to kill off the hardcoded inv1/2/3 or are they too baked into borg existence

so much code duplication

File: code/modules/mob/living/silicon/damage_procs.dm [hunk 2]
```diff
@@ -30,14 +27,26 @@
...
 	return FALSE
 
 /mob/living/silicon/setOrganLoss(slot, amount)
 	return FALSE
+
+/mob/living/silicon/adjustOxyLoss(amount, updating_health = TRUE, forced = FALSE) //immune to oxygen damage
+	if(istype(src, /mob/living/silicon/ai)) //ais are snowflakes and use oxyloss for being in AI cards and having no battery
+		. = ..()
+	
+	return FALSE
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/damage_procs.dm:46]: this isn't returning the proper value for AI mobs.

File: code/modules/mob/living/silicon/damage_procs.dm [hunk 3]
```diff
@@ -30,14 +27,26 @@
...
+
+/mob/living/silicon/setOxyLoss(amount, updating_health = TRUE, forced = FALSE)
+	if(istype(src, /mob/living/silicon/ai)) //ditto
+		. = ..()
+	
+	return FALSE
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/damage_procs.dm:52]: same

File: code/modules/mob/living/silicon/damage_procs.dm [hunk 4]
```diff
@@ -30,14 +27,26 @@
...
-/mob/living/silicon/adjustOrganLoss(slot, amount, maximum = 500)
+/mob/living/silicon/adjustOrganLoss(slot, amount, maximum = 500) //immune to organ damage (no organs, duh)
 	return FALSE
 
 /mob/living/silicon/setOrganLoss(slot, amount)
 	return FALSE
+
+/mob/living/silicon/adjustOxyLoss(amount, updating_health = TRUE, forced = FALSE) //immune to oxygen damage
+	if(istype(src, /mob/living/silicon/ai)) //ais are snowflakes and use oxyloss for being in AI cards and having no battery
+		. = ..()
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/damage_procs.dm:44]: return ..()

File: code/modules/mob/living/silicon/damage_procs.dm [hunk 5]
```diff
@@ -30,14 +27,26 @@
...
+	
+	return FALSE
+
+/mob/living/silicon/setOxyLoss(amount, updating_health = TRUE, forced = FALSE)
+	if(istype(src, /mob/living/silicon/ai)) //ditto
+		. = ..()
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/damage_procs.dm:50]: return ..()

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 6]
```diff
@@ -1,82 +1,233 @@
...
+	if(activated(I))
+		to_chat(src, "<span class='warning'>That module is already activated.</span>")
+		return FALSE
+	
+	if(disabled_modules & BORG_MODULE_ALL_DISABLED)
+		to_chat(src, "<span class='warning'>All modules are disabled!</span>")
+		return FALSE
+
+	/// What's the first free slot for the borg?
+	var/first_free_slot = 0
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:38]: var/first_free_slot = !held_items[1] ? 1 : (!held_items[2] ? 2 : (!held_items[3] ? 3 : null)

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 7]
```diff
@@ -92,124 +243,139 @@
...
+  * Checks if the provided module number is a valid number.
+  * 
+  * If the number is between 1 and 3, or between 1 and the number of disabled
+  * modules (if check_all_slots is true), then it returns FALSE. Otherwise, it returns TRUE.
+  * Arguments
+  * * module_num - the passed module num that is checked for validity.
+  * * check_all_slots - TRUE = the proc checks all slots | FALSE = the proc only checks un-disabled slots
+  */
+/mob/living/silicon/robot/proc/is_invalid_module_number(module_num, check_all_slots = FALSE)
+	/// The number of module slots we're checking
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:282]:
```
if(!module_num)
	return false
```

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 8]
```diff
@@ -1,82 +1,233 @@
...
+			disabled_modules |= BORG_MODULE_TWO_DISABLED
+
+		if(3)
+			if(disabled_modules & BORG_MODULE_THREE_DISABLED)
+				return FALSE
+		
+			inv3.icon_state = "inv3 +b"
+			disabled_modules |= BORG_MODULE_THREE_DISABLED
+
+	playsound(loc, 'sound/machines/warning-buzzer.ogg', 50, TRUE, TRUE)
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:170]: src

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 9]
```diff
@@ -1,82 +1,233 @@
...
+	audible_message("<span class='warning'>[src] sounds an alarm! \"SYSTEM ERROR: Module [module_num] OFFLINE.\"</span>", self_message = null)
+	to_chat(src, "<span class='userdanger'>SYSTEM ERROR: Module [module_num] OFFLINE.</span>")
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:172]:
```
 					audible_message("<span class='warning'>[src] sounds an alarm! \"CRITICAL ERROR: All modules OFFLINE.\"</span>")
					to_chat(src, "<span class='userdanger'>CRITICAL ERROR: All modules OFFLINE.</span>")
					playsound(loc, 'sound/machines/warning-buzzer.ogg', 75, TRUE, TRUE)
```

preserve the old functionality of these getting louder with each one that gets disabled / a message saying everything is disabled when everything is... well, disabled.

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 10]
```diff
@@ -1,82 +1,233 @@
...
-	if(O.item_flags & DROPDEL)
-		O.item_flags &= ~DROPDEL //we shouldn't HAVE things with DROPDEL_1 in our modules, but better safe than runtiming horribly
-
-	O.forceMove(module) //Return item to module so it appears in its contents, so it can be taken out again.
-	O.cyborg_unequip(src)
 
+	switch(module_num)
+		if(1)
+			if(!(disabled_modules & BORG_MODULE_ALL_DISABLED))
+				inv1.icon_state = "inv1"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:115]: initial(icon_state)

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 11]
```diff
@@ -1,82 +1,233 @@
...
+		if(2)
+			if(!(disabled_modules & BORG_MODULE_TWO_DISABLED))
+				inv2.icon_state = "inv2"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:118]: initial(icon_state)

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 12]
```diff
@@ -1,82 +1,233 @@
...
+		if(3)
+			if(!(disabled_modules & BORG_MODULE_THREE_DISABLED))
+				inv3.icon_state = "inv3"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:121]: initial(icon_state)

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 13]
```diff
@@ -1,82 +1,233 @@
...
+	if(held_items[module_num]) //If there's a held item, unequip it first.
+		if(!unequip_module_from_slot(held_items[module_num], module_num)) //If we fail to unequip it, then don't continue
+			return FALSE
+
+	switch(module_num)
+		if(1)
+			if(disabled_modules & BORG_MODULE_ALL_DISABLED)
+				return FALSE
+		
+			inv1.icon_state = "inv1 +b"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:153]: "[initial(inv1.icon_state)] +b"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 14]
```diff
@@ -1,82 +1,233 @@
...
+			disabled_modules |= BORG_MODULE_ALL_DISABLED
 
+		if(2)
+			if(disabled_modules & BORG_MODULE_TWO_DISABLED)
+				return FALSE
+		
+			inv2.icon_state = "inv2 +b"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:160]: "[initial(inv2.icon_state)] +b"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 15]
```diff
@@ -1,82 +1,233 @@
...
+			disabled_modules |= BORG_MODULE_TWO_DISABLED
+
+		if(3)
+			if(disabled_modules & BORG_MODULE_THREE_DISABLED)
+				return FALSE
+		
+			inv3.icon_state = "inv3 +b"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:167]: "[initial(inv3.icon_state)] +b"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 16]
```diff
@@ -1,82 +1,233 @@
...
+/mob/living/silicon/robot/proc/repair_cyborg_slot(module_num)
+	if(is_invalid_module_number(module_num, TRUE))
+		return FALSE
 
+	switch(module_num)
+		if(1)
+			if(!(disabled_modules & BORG_MODULE_ALL_DISABLED))
+				return FALSE
+		
+			inv1.icon_state = "inv1"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:199]: "[initial(inv1.icon_state)]"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 17]
```diff
@@ -1,82 +1,233 @@
...
+			disabled_modules &= ~BORG_MODULE_ALL_DISABLED
+		if(2)
+			if(!(disabled_modules & BORG_MODULE_TWO_DISABLED))
+				return FALSE
+		
+			inv2.icon_state = "inv2"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:205]: "[initial(inv2.icon_state)]"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 18]
```diff
@@ -1,82 +1,233 @@
...
+			disabled_modules &= ~BORG_MODULE_TWO_DISABLED
+		if(3)
+			if(!(disabled_modules & BORG_MODULE_THREE_DISABLED))
+				return FALSE
+		
+			inv3.icon_state = "inv3"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:211]: "[initial(inv3.icon_state)]"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 19]
```diff
@@ -92,124 +243,139 @@
...
+/mob/living/silicon/robot/proc/select_module(module_num)
+	if(!held_items[module_num])
+		return FALSE
 
-	switch(module)
+	switch(module_num)
 		if(1)
-			if(module_active != held_items[module])
+			if(module_active != held_items[module_num])
 				inv1.icon_state = "inv1 +a"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:315]: "[initial(inv1.icon_state)] +a"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 20]
```diff
@@ -92,124 +243,139 @@
...
-				inv2.icon_state = "inv2"
-				inv3.icon_state = "inv3"
 		if(2)
-			if(module_active != held_items[module])
-				inv1.icon_state = "inv1"
+			if(module_active != held_items[module_num])
 				inv2.icon_state = "inv2 +a"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:318]: "[initial(inv2.icon_state)] +a"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 21]
```diff
@@ -92,124 +243,139 @@
...
-				inv3.icon_state = "inv3"
 		if(3)
-			if(module_active != held_items[module])
-				inv1.icon_state = "inv1"
-				inv2.icon_state = "inv2"
+			if(module_active != held_items[module_num])
 				inv3.icon_state = "inv3 +a"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:321]: "[initial(inv3.icon_state)] +a"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 22]
```diff
@@ -92,124 +243,139 @@
...
+  * Deselects the module in the slot module_num.
+  * Arguments
+  * * module_num - the slot number being de-selected
+  */
+/mob/living/silicon/robot/proc/deselect_module(module_num)
+	switch(module_num)
 		if(1)
-			if(module_active == held_items[module])
+			if(module_active == held_items[module_num])
 				inv1.icon_state = "inv1"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:333]: "[initial(inv1.icon_state)]"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 23]
```diff
@@ -92,124 +243,139 @@
...
 		if(2)
-			if(module_active == held_items[module])
+			if(module_active == held_items[module_num])
 				inv2.icon_state = "inv2"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:336]: "[initial(inv2.icon_state)]"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 24]
```diff
@@ -92,124 +243,139 @@
...
 		if(3)
-			if(module_active == held_items[module])
+			if(module_active == held_items[module_num])
 				inv3.icon_state = "inv3"
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:339]: "[initial(inv3.icon_state)]"

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 25]
```diff
@@ -1,82 +1,233 @@
...
+	/// What's the first free slot for the borg?
+	var/first_free_slot = 0
+	if(!held_items[1])
+		first_free_slot = 1
+	else if(!held_items[2])
+		first_free_slot = 2
+	else if(!held_items[3])
+		first_free_slot = 3
+
+	if(is_invalid_module_number(first_free_slot))
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:46]: if(!first_free_slot || is_invalid_module_number(first_free_slot))

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 26]
```diff
@@ -1,82 +1,235 @@
...
+
+		if(3)
+			if(disabled_modules & BORG_MODULE_THREE_DISABLED)
+				return FALSE
+
+			inv3.icon_state = "[initial(inv3.icon_state)] +b"
+			disabled_modules |= BORG_MODULE_THREE_DISABLED
+
+			playsound(src, 'sound/machines/warning-buzzer.ogg', 75, TRUE, TRUE)
+			audible_message("<span class='warning'>[src] sounds an alarm! \"CRITICAL ERROR: ALL modules OFFLINE.\"</span>")
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:173]: 1 & 3 are reversed.

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 27]
```diff
@@ -1,82 +1,235 @@
...
+			to_chat(src, "<span class='userdanger'>CRITICAL ERROR: ALL modules OFFLINE.</span>")
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:174]: .

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 28]
```diff
@@ -1,82 +1,235 @@
...
+			to_chat(src, "<span class='userdanger'>SYSTEM ERROR: Module [module_num] OFFLINE.</span>")
+
+		if(3)
+			if(disabled_modules & BORG_MODULE_THREE_DISABLED)
+				return FALSE
+
+			inv3.icon_state = "[initial(inv3.icon_state)] +b"
+			disabled_modules |= BORG_MODULE_THREE_DISABLED
+
+			playsound(src, 'sound/machines/warning-buzzer.ogg', 75, TRUE, TRUE)
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:172]: .

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 29]
```diff
@@ -1,82 +1,235 @@
...
+
+	switch(module_num)
+		if(1)
+			if(disabled_modules & BORG_MODULE_ALL_DISABLED)
+				return FALSE
+
+			inv1.icon_state = "[initial(inv1.icon_state)] +b"
+			disabled_modules |= BORG_MODULE_ALL_DISABLED
 
+			playsound(src, 'sound/machines/warning-buzzer.ogg', 50, TRUE, TRUE)
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:150]: .

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 30]
```diff
@@ -1,82 +1,235 @@
...
+			audible_message("<span class='warning'>[src] sounds an alarm! \"SYSTEM ERROR: Module [module_num] OFFLINE.\"</span>")
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:151]: .

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 31]
```diff
@@ -1,82 +1,235 @@
...
+			to_chat(src, "<span class='userdanger'>SYSTEM ERROR: Module [module_num] OFFLINE.</span>")
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/robot/inventory.dm:152]: .

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Review by Twaticus — APPROVED** [review]

**Comment by zombiegums** [comment]
> In the process, this fixes the use-items-in-broken-slots tech. Sorry, borg mains.

:'(

On a serious note, thank you for tidying things up!

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 32]
```diff
@@ -1,82 +1,235 @@
...
+		return FALSE
+
+	/// What's the first free slot for the borg?
+	var/first_free_slot = !held_items[1] ? 1 : (!held_items[2] ? 2 : (!held_items[3] ? 3 : null))
+
+	if(!first_free_slot || is_invalid_module_number(first_free_slot))
+		to_chat(src, "<span class='warning'>Deactivate a module first!</span>")
+		return FALSE
+
+	return equip_module_to_slot(I, first_free_slot)
```
Comment (Rohesie) [inline code/modules/mob/living/silicon/robot/inventory.dm:44]:
```suggestion
/mob/living/silicon/robot/proc/activate_module(obj/item/item_module)
	if(QDELETED(item_module))
		CRASH("activate_module called with improper item_module")

	if(!(item_module in module.modules))
		CRASH("activate_module called with item_module not in module.modules")

	if(activated(item_module))
		to_chat(src, "<span class='warning'>That module is already activated.</span>")
		return FALSE

	if(disabled_modules & BORG_MODULE_ALL_DISABLED)
		to_chat(src, "<span class='warning'>All modules are disabled!</span>")
		return FALSE

	/// What's the first free slot for the borg?
	var/first_free_slot = !held_items[1] ? 1 : (!held_items[2] ? 2 : (!held_items[3] ? 3 : null))

	if(!first_free_slot || is_invalid_module_number(first_free_slot))
		to_chat(src, "<span class='warning'>Deactivate a module first!</span>")
		return FALSE

	return equip_module_to_slot(item_module, first_free_slot)
```

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 33]
```diff
@@ -1,82 +1,235 @@
...
+	I.plane = ABOVE_HUD_PLANE
+	I.forceMove(src)
+
+	if(istype(I, /obj/item/borg/sight))
+		var/obj/item/borg/sight/S = I
+		sight_mode |= S.sight_mode
+		update_sight()
+
+	observer_screen_update(I,TRUE)
+	return TRUE
```
Comment (Rohesie) [inline code/modules/mob/living/silicon/robot/inventory.dm:75]:
```suggestion
/mob/living/silicon/robot/proc/equip_module_to_slot(obj/item/item_module, module_num)
	switch(module_num)
		if(1)
			item_module.screen_loc = inv1.screen_loc
		if(2)
			item_module.screen_loc = inv2.screen_loc
		if(3)
			item_module.screen_loc = inv3.screen_loc

	held_items[module_num] = item_module
	item_module.equipped(src, ITEM_SLOT_HANDS)
	item_module.mouse_opacity = initial(I.mouse_opacity)
	item_module.layer = ABOVE_HUD_LAYER
	item_module.plane = ABOVE_HUD_PLANE
	item_module.forceMove(src)

	if(istype(item_module, /obj/item/borg/sight))
		var/obj/item/borg/sight/borg_sight = item_module
		sight_mode |= borg_sight.sight_mode
		update_sight()

	observer_screen_update(item_module, TRUE)
	return TRUE
```

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 34]
```diff
@@ -92,124 +245,142 @@
...
+  */
 /mob/living/silicon/robot/proc/uneq_all()
-	for(var/obj/item/I in held_items)
-		uneq_module(I)
-
-/mob/living/silicon/robot/proc/activated(obj/item/O)
-	if(O in held_items)
+	for(var/I in 1 to 3)
+		if(held_items[I])
+			unequip_module_from_slot(held_items[I], I)
```
Comment (Rohesie) [inline code/modules/mob/living/silicon/robot/inventory.dm:260]:
```suggestion
	for(var/item_module in 1 to 3)
		if(!held_items[item_module])
			continue
		unequip_module_from_slot(held_items[item_module], item_module)
```

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 35]
```diff
@@ -92,124 +245,142 @@
...
+
+/**
+  * Checks if the item is currently in a slot.
+  *
+  * If the item is found in a slot, this returns TRUE. Otherwise, it returns FALSE
+  * Arguments
+  * * I - the item being checked
+  */
+/mob/living/silicon/robot/proc/activated(obj/item/I)
+	if(I in held_items)
```
Comment (Rohesie) [inline code/modules/mob/living/silicon/robot/inventory.dm:270]:
```suggestion
/mob/living/silicon/robot/proc/activated(obj/item/item_module)
	if(item_module in held_items)
```

**Comment by carshalash** [comment]
This is INCREDIBLY buggy and makes hotkey switching as a cyborg a nightmare as well as just 100% inaccuracy on which item you're using if you're using all three slots.

**Comment by MrMelbert** [comment]
@carshalash i never had any problem testing, can you share the process in which you broke it / record you breaking it?

**Comment by carshalash** [comment]
Cycling is 100% busted.

image
First point, all items, nothing selected yet
image
Smack cycle button once, EVERYTHING lights up, wrench is the item that is equipped instead of the welding tool
image
Switches to the screwdriver because? Other 2 items are lit up
image
Welder is in use now, others stay lit up.

Cycling is in reverse for, some reason and other items constantly stay lit up while cycling.

**Comment by bobbahbrown** [comment]
tested code

**Comment by MrMelbert** [comment]
just a bug with cycling, or anything else? the cycling should be an easy fix cause I think I know where that went wrong

**Comment by carshalash** [comment]
Seems to be that from what i've found so far, it is VERY bad for cycling.

**Comment by MrMelbert** [comment]
okay, i'll fix the cycling in a bit, at least it's not completely awful like preventing borgs from using modules (which is what I was worried would be a bug by the way you phrased it)

**Pull request merged**